src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblySerializationModeHandler.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Services;
+
+internal class WebAssemblySerializationModeHandler : ISerializationModeHandler
+{
+    public PersistedStateSerializationMode GlobalSerializationMode
+    {
+        get => PersistedStateSerializationMode.WebAssembly;
+        set => throw new NotImplementedException("Cannot change global serialization mode.");
+    }
+
+    public PersistedStateSerializationMode GetCallbackTargetSerializationMode(object? callbackTarget)
+        => PersistedStateSerializationMode.WebAssembly;
```

Comment:
I'm failing to understand what this is doing if it's just returning a constant instead of looking at the callback target.

Reply:
Well, short answer to your question is because for webassembly the callback target can only be `PersistedStateSerializationMode.WebAssembly`.

I guess the confusing part is why there are 3 `ISerializationModeHandler`s: `EndpointHtmlRenderer`, `ServerSerializationModeHandler`, `WebAssemblySerializationModeHandler` (actually now listing them, I realised it might be better to add `EndpointSerializationModeHandler` and use `EndpointHtmlRenderer` inside of it).

Reply:
Well, short answer to your question is because for webassembly the callback target can only be `PersistedStateSerializationMode.WebAssembly`.

I guess the confusing part is why there are 3 `ISerializationModeHandlers`: `EndpointHtmlRenderer`, `ServerSerializationModeHandler`, `WebAssemblySerializationModeHandler` (actually now listing them, I realised it might be better to add `EndpointSerializationModeHandler` and use `EndpointHtmlRenderer` inside of it).

**EndpointHtmlRenderer**
`GetCallbackTargetSerializationMode` checks if the callback target is a component and then gets the component's render mode. 
However this doesn't work on the circuit because the component's render mode is not available on the circuit. The reason is that the circuit doesn't use the same instance of `PersistentComponentState` and `ISerializationModeHandlers`, it has its own scope. The circuit uses same exact kinds of services listed in `AddRazorComponents` though.
Hence, there is a need to be able to change the `GlobalSerializationMode` to `Server` when creating the circuit host [here](https://github.com/dotnet/aspnetcore/pull/50373/files/7bbfde083b1f483cd2fc3686cbec1e1f41420281#diff-4a5e6027a7a100cacb4a0f69d0c548e7c01379898af20a265297840d171f38b0R66).


**WebAssemblySerializationModeHandler**
The webassembly host uses the services provided [here](https://github.com/dotnet/aspnetcore/pull/50373/files/7bbfde083b1f483cd2fc3686cbec1e1f41420281#diff-e5845021124dd8ca74320ce8dbc62fa89ed95f3a7addf44ed99120a9d501a279R254). 
Thant is why I created `WebAssemblySerializationModeHandler` that will just return constant `PersistedStateSerializationMode.WebAssembly` and throws when trying to set `GlobalSerializationMode`.


**ServerSerializationModeHandler**
This one is for pure Blazor Server apps. I baked it [here](https://github.com/dotnet/aspnetcore/pull/50373/files#diff-c1d00e1ea2b0628c9c2d2a32c1e59e625dd3c972e9938b4b0f33d2709e9ec5bfR212) before `AddRazorComponents` so it would override `EndpointHtmlRenderer`. This seemed like a right place after following AddRazor**Pages**.


Hope this answers your question.